akka-actor-tests/src/test/scala/akka/pattern/BackoffSupervisorSpec.scala

```diff
@@ -70,5 +85,39 @@ class BackoffSupervisorSpec extends AkkaSpec with ImplicitSender {
         expectMsgType[BackoffSupervisor.CurrentChild].ref.get should !==(c1)
       }
     }
+
+    "support manual reset" in {
```

Comment:
Maybe parameterize this whole block so that both onStop() and onFailure() can both be tested